### PR TITLE
[Example] Add EGES example

### PR DESCRIPTION
**Pull request by Wang-Yu-Qing**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [ ] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by BarclayII** [comment]
Could you remove the CSV files and put a URL in the README for downloading (as well as where to put them) instead?

**Comment by Wang-Yu-Qing** [comment]
> Could you remove the CSV files and put a URL in the README for downloading (as well as where to put them) instead?

I've removed the CSV files and added a data link in the README.

File: examples/pytorch/eges/main.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+
+
+
+
+
+    
+
+    
+
```
Comment (jermainewang) [inline examples/pytorch/eges/main.py:110]: Remove all these trailing newlines.

File: examples/pytorch/eges/model.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        dots = th.sigmoid(th.sum(srcs * dsts, axis=1))
+        dots = th.clamp(dots, min=1e-7, max=1 - 1e-7)
+
+        return th.mean(- (labels * th.log(dots) + (1 - labels) * th.log(1 - dots)))
+
+    def query_cold_item(self):
+        pass
+    
+
+
```
Comment (jermainewang) [inline examples/pytorch/eges/model.py:59]: Same here

File: examples/pytorch/eges/utils.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        test_edges.append(TestEdge(src, dst, 0))
+    
+    graph.remove_edges(sampled_edge_ids)
+    test_graph = test_edges
+
+    return graph, test_graph
+
+    
+           
+
```
Comment (jermainewang) [inline examples/pytorch/eges/utils.py:219]: same here

Comment (Wang-Yu-Qing) [reply examples/pytorch/eges/utils.py]: Hi, I've removed all these trailing newlines.

**Review by jermainewang — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**